tools/gyp/pylib/gyp/input.py

```diff
@@ -907,11 +899,13 @@ def ExpandVariables(input, phase, variables, build_file):
                                  stderr=subprocess.PIPE,
                                  stdin=subprocess.PIPE,
                                  cwd=build_file_dir)
-          except Exception, e:
+          except Exception as e:
             raise GypError("%s while executing command '%s' in %s" %
                            (e, contents, build_file))
 
           p_stdout, p_stderr = p.communicate('')
+          p_stdout = to_utf8(p_stdout)
+          p_stderr = to_utf8(p_stderr)
```

Comment:
I suppose this might cause a runtime exception when the output contains e.g. latin1-encoded file paths?

Reply:
You are correct.  There is the old adage that you can never accurately guess encodings.  How should we resolve?  We could catch the __UnicodeDecodeError__ and then attempt __s.decode('latin-1')__ and it we catch another  __UnicodeDecodeError__ just return __s__.  Other approaches?

Reply:
That sounds reasonable, that probably covers 99% of everyday use.